src/ctx.cpp

```diff
@@ -1979,13 +1979,26 @@ llvm::Instruction *FunctionEmitContext::ZExtInst(llvm::Value *value, llvm::Type
     return inst;
 }
 
+/** Extend both uniform and varying integer values to 64 bits. */
+llvm::Value *lExtendIntTo64(FunctionEmitContext *ctx, llvm::Value *value, const Type *indexType,
+                            const llvm::Twine &name = "") {
+    bool valueIsVarying = llvm::isa<llvm::VectorType>(value->getType());
```

Comment:
We have this function `IsVaryingAtomicOrUniformVectorType`. Is it what you want to check here?